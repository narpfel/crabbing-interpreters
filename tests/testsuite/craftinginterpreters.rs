use std::path::Path;
use std::path::PathBuf;
use std::process::Command;

use insta_cmd::assert_cmd_snapshot;
use insta_cmd::get_cargo_bin;
use rstest::rstest;

fn as_relative(path: &Path) -> &Path {
    path.strip_prefix(env!("CARGO_MANIFEST_DIR")).unwrap()
}

#[rstest]
fn tests(#[files("craftinginterpreters/test/operator/**/*.lox")] path: PathBuf) {
    let path = as_relative(&path);
    assert_cmd_snapshot!(
        path.display().to_string(),
        Command::new(get_cargo_bin("crabbing-interpreters")).arg(path)
    )
}
