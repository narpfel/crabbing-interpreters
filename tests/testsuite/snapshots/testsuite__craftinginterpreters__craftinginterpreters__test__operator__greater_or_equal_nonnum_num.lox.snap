---
source: tests/testsuite/craftinginterpreters.rs
info:
  program: crabbing-interpreters
  args:
    - craftinginterpreters/test/operator/greater_or_equal_nonnum_num.lox
---
success: false
exit_code: 1
----- stdout -----

----- stderr -----
InvalidBinaryOp { lhs: String("1"), op: BinOp { kind: GreaterEqual, token: Token { kind: GreaterEqual, loc: craftinginterpreters/test/operator/greater_or_equal_nonnum_num.lox:4..6 } }, rhs: Number(1.0), at: Binary { lhs: Literal(Literal { kind: String("1"), token: Token { kind: String, loc: craftinginterpreters/test/operator/greater_or_equal_nonnum_num.lox:0..3 } }), op: BinOp { kind: GreaterEqual, token: Token { kind: GreaterEqual, loc: craftinginterpreters/test/operator/greater_or_equal_nonnum_num.lox:4..6 } }, rhs: Literal(Literal { kind: Number(1.0), token: Token { kind: Number, loc: craftinginterpreters/test/operator/greater_or_equal_nonnum_num.lox:7..8 } }) } }
